omegaconf/built_in_resolvers.py

```diff
@@ -69,3 +84,26 @@ def legacy_env(key: str, default: Optional[str] = None) -> Any:
             return decode_primitive(default)
         else:
             raise ValidationError(f"Environment variable '{key}' not found")
+
+
+def _dict_impl(
+    keys_or_values: str,
+    in_dict: Union[str, Mapping[Any, Any]],
+    _root_: BaseContainer,
+    _parent_: Container,
+) -> ListConfig:
+    if isinstance(in_dict, str):
+        # Path to an existing key in the config: use `select()`.
+        in_dict = OmegaConf.select(_root_, in_dict, throw_on_missing=True)
+
+    if not isinstance(in_dict, Mapping):
+        raise TypeError(
+            f"`oc.dict.{keys_or_values}` cannot be applied to objects of type: "
+            f"{type(in_dict).__name__}"
+        )
+
+    dict_method = getattr(in_dict, keys_or_values)
+    assert isinstance(_parent_, BaseContainer)
+    ret = OmegaConf.create(list(dict_method()), parent=_parent_)
+    assert isinstance(ret, ListConfig)
+    return ret
```

Comment:
IMO would be cleaner to return the dict/DictConfig object and allow the caller to just call the appropriate function.

Reply:
Done in cdc60a8